### PR TITLE
Force backends to register by calling dlopen

### PR DESCRIPTION
**Pull request by rkimballn1**
If a user requests a backend like this `Backend::create("IE")` and that backend has not been registered yet then try opening it via dlopen which will force it to register.
We try the following name combinations for the name of the shared library:
```
libIE.so
libIE_backend.so
libie.so
libie_backend.so
```
If any of those work then the backend will register itself and all is well.

### CODE REVIEW

File: src/ngraph/runtime/backend.cpp [hunk 1]
```diff
@@ -39,12 +40,41 @@ runtime::Backend::~Backend()
 {
 }
 
+void* runtime::Backend::open_shared_library(const string& type)
+{
+    void* handle = nullptr;
+    vector<string> names;
+    names.push_back("lib" + type + ".so");
```
Comment (diyessi) [inline src/ngraph/runtime/backend.cpp]: Why do we need all this DWIM for names? Can't we just define what the name needs to look like and enforce it?

Comment (rkimballn1) [reply src/ngraph/runtime/backend.cpp]: Sure. What should we define it to be?

**Comment by rkimballn1** [comment]
TESTNOW

**Comment by rkimballn1** [comment]
TESTNOW

**Comment by rkimballn1** [comment]
TESTNOW

**Comment by rkimballn1** [comment]
TESTNOW

**Comment by rkimballn1** [comment]
TESTNOW

**Pull request merged**